### PR TITLE
chore(changelog): Announce endpoint hardening

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

This change announces a change of behaviour for some product endpoints.
This is a Changelog only modification.

## Checklist
- [ ] Investigated and inspected CI test results
~~- [ ] Unit test and regression tests added~~
- [ ] Evaluated and added CHANGELOG entry if required
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI should be enough as only the Changelog is impacted.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
 - The removal of `/v1/report` APIs in this release, that was communicated in release 4.0.0, has been postponed by one release. Consequently, the `/v1/report` APIs will continue to be available in this release.
 - The `/api/docs/swagger` API previously required read on the resource `Integration`.
   Now it only requires users to be authenticated to via the API docs.
+- ROX-18173: A few public endpoints will soon require authentication: `/v1/availableauthproviders`, `/v1/config/public`, `/v1/featureflags`, `/v1/internal/dev/certcheck`, `/v1/internal/dev/envvars`, `/v1/internal/dev/random`, `/v1/internal/dev/reconciliationstats`, `/v1/internal/dev/replicate` and `/v1/resources`. Ensure the flows targeting these endpoints are authenticated.
```
Comment (dhaus67) [inline CHANGELOG.md:53]: nit: make this a multi-line listing of the endpoints?

Also, the remediation can be re-written to `Ensure that any flow interacting with these endpoints is authenticated going forward`.

**Review by dhaus67 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -50,6 +50,16 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
 - The removal of `/v1/report` APIs in this release, that was communicated in release 4.0.0, has been postponed by one release. Consequently, the `/v1/report` APIs will continue to be available in this release.
 - The `/api/docs/swagger` API previously required read on the resource `Integration`.
   Now it only requires users to be authenticated to via the API docs.
+- ROX-18173: A few public endpoints will soon require authentication, ensure that any flow interacting with these endpoints is authenticated going forward:
+  - `/v1/availableauthproviders`
```
Comment (rukletsov) [inline CHANGELOG.md:54]: Doubt this will become authenticated, this is needed for UI login screen

File: CHANGELOG.md [hunk 3]
```diff
@@ -50,6 +50,16 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
...
+  - `/v1/config/public`
```
Comment (rukletsov) [inline CHANGELOG.md:55]: This already requires authn, which is incorrect : ).

File: CHANGELOG.md [hunk 4]
```diff
@@ -50,6 +50,16 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
...
+  - `/v1/featureflags`
+  - `/v1/internal/dev/certcheck`
+  - `/v1/internal/dev/envvars`
+  - `/v1/internal/dev/random`
+  - `/v1/internal/dev/reconciliationstats`
+  - `/v1/internal/dev/replicate`
```
Comment (rukletsov) [inline CHANGELOG.md:61]: I'm not sure about these ones. They are actually not part of the release build. Let's ditch them for now.

**Review by rukletsov — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -50,6 +50,16 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
 - The removal of `/v1/report` APIs in this release, that was communicated in release 4.0.0, has been postponed by one release. Consequently, the `/v1/report` APIs will continue to be available in this release.
 - The `/api/docs/swagger` API previously required read on the resource `Integration`.
   Now it only requires users to be authenticated to via the API docs.
+- ROX-18173: A few public endpoints will soon require authentication, ensure that any flow interacting with these endpoints is authenticated going forward:
```
Comment (rukletsov) [inline CHANGELOG.md:53]: Also, ditch the ticket and move to ### Deprecated Features

File: CHANGELOG.md (hunk 2 above)
Comment (rhybrillou) [reply CHANGELOG.md:54]: Removed for now.

**Pull request merged**